portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,7 +36,49 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Chris!</h1>");
+    // Prepare query.	 
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    int maxComments = Integer.parseInt(request.getParameter("maxComments"));		
+    
+    // Get desired number of comments.		
+    List<String> comments = new ArrayList<>();
+    int numComments = 0;
```

Comment:
I think you might be able to use `comments.size()` to count the number of comments you've read, and not need to keep a separate `numComments` variable

Reply:
Thanks, fixed